dbt-snowflake/src/dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -31,14 +31,16 @@
         A valid DDL statement which will result in a new dynamic info schema table.
 -#}
 
-create dynamic table {{ relation }}
-    target_lag = '{{ dynamic_table.target_lag }}'
-    warehouse = {{ dynamic_table.snowflake_warehouse }}
-    {{ optional('refresh_mode', dynamic_table.refresh_mode) }}
-    {{ optional('initialize', dynamic_table.initialize) }}
-    as (
-        {{ sql }}
-    )
+    create dynamic table {{ relation }}
+        target_lag = '{{ dynamic_table.target_lag }}'
+        warehouse = {{ dynamic_table.snowflake_warehouse }}
+        {{ optional('refresh_mode', dynamic_table.refresh_mode) }}
+        {{ optional('initialize', dynamic_table.initialize) }}
+        {{ optional('with row access policy', dynamic_table.row_access_policy, with_equal=false) }}
+        {{ optional('with tag', dynamic_table.table_tag, quote_char='(', quote_char_right=')', with_equal=false) }}
```

Comment:
There's no `quote_char_right` argument to `optional`. It will use `')'` for the end quote if `quote_char='('`.

There's no `with_equal` argument to `optional`. To unset the equal character, supply `equals_char=''`.

Reply:
fixed